api/controllers/course.go

```diff
@@ -35,8 +37,22 @@ func CourseSearch() gin.HandlerFunc {
 			query[key] = c.Query(key)
 		}
 
+		delete(query, "offset") // offset not in query because it is for pagination not searching
```

Comment:
Is "offset" guaranteed to be in query?

Reply:
No. If "offset" is not in query, then no change occurs when performing delete(query, "offset).